docs/source/inference.rst

```diff
@@ -0,0 +1,189 @@
+
+    # Using PyTorch Backend
+    dglke_score --data_path data/wn18/ --model_path ckpts/DistMult_wn18_0/ --format 'h_r_t' --data_files head.list rel.list tail.list --score_func none --topK 5
+
+    # Using MXNet Backend
+    MXNET_ENGINE_TYPE=NaiveEngine DGLBACKEND=mxnet dglke_score --data_path data/wn18/ --model_path ckpts/DistMult_wn18_0/ --format 'h_r_t' --data_files head.list rel.list tail.list --score_func none --topK 5
+
+The output is as::
+
+    src  rel  dst  score
+    6    0    15   -2.39380
+    8    0    14   -2.65297
+    2    0    14   -2.67331
+    9    0    18   -2.86985
+    8    0    20   -2.89651
+
+The following command shows how to do linkage score ranking while broadcasting at head using a pretrained TransE_l2 model::
+
+    # Using PyTorch Backend
+    dglke_score --data_path data/wn18/ --model_path ckpts/TransE_l2_wn18_0/ --format 'h_r_t' --data_files head.list rel.list tail.list --score_func logsigmoid --topK 5 --bcast head
+
+    # Using MXNet Backend
+    MXNET_ENGINE_TYPE=NaiveEngine DGLBACKEND=mxnet dglke_score --data_path data/wn18/ --model_path ckpts/TransE_l2_wn18_0/ --format 'h_r_t' --data_files head.list rel.list tail.list --score_func logsigmoid --topK 5 --bcast head
+
+The output is as::
+
+    src  rel  dst  score
+    1    0    12   -5.11393
+    1    0    18   -6.10925
+    1    0    13   -6.66778
+    1    0    17   -6.81532
+    1    0    19   -6.83329
+    2    0    17   -5.09325
+    2    0    18   -5.42972
+    2    0    20   -5.61894
+    2    0    12   -5.75848
+    2    0    14   -5.94183
+    ...
+
+The following command shows how to do linkage score ranking using a pretrained TransE_l2 model and use raw ID space (turn on --raw_data)::
+
+    # Using PyTorch Backend
+    dglke_score --data_path data/wn18/ --model_path ckpts/TransE_l2_wn18_0/ --format 'h_r_t' --data_files raw_head.list raw_rel.list raw_tail.list --topK 5 --raw_data
+
+    # Using MXNet Backend
+    MXNET_ENGINE_TYPE=NaiveEngine DGLBACKEND=mxnet dglke_score --data_path data/wn18/ --model_path ckpts/TransE_l2_wn18_0/ --format 'h_r_t' --data_files raw_head.list raw_rel.list raw_tail.list --topK 5 --raw_data
+
+The output is as::
+
+    head      rel                           tail      score
+    08847694  _derivationally_related_form  09440400  -7.41088
+    08847694  _hyponym                      09440400  -8.99562
+    02537319  _derivationally_related_form  01490112  -9.08666
+    02537319  _hyponym                      01490112  -9.44877
+    00083809  _derivationally_related_form  05940414  -9.88155
+
+The Embedding Similarity Ranking
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The task of embedding similarity ranking is given a list of entity (e1, e2) pairs or relation (r1, r2) pairs, calculating the similarity between their corresponding embeddings and returning the topk most similar pairs. An example of return value of top5 similar entities likes this::
+
+    head  tail  score
```

Comment:
is this actual head entity?

Reply:
Emmm..., the naming here is miss leading. It should be emb1 and emb2, not the head and tail in the triplets.

Reply:
yes, i think we should call them entity1 and entity2.